common/desktop-exports

```diff
@@ -79,51 +79,55 @@ append_dir PATH "$SNAP_DESKTOP_RUNTIME/usr/bin"
 # XKB config
 export XKB_CONFIG_ROOT="$SNAP_DESKTOP_RUNTIME/usr/share/X11/xkb"
 
-# Give XOpenIM a chance to locate locale data.
-# This is required for text input to work in SDL2 games.
-export XLOCALEDIR="$SNAP_DESKTOP_RUNTIME/usr/share/X11/locale"
-
 # Set XCursors path
 export XCURSOR_PATH="$SNAP_DESKTOP_RUNTIME/usr/share/icons"
 prepend_dir XCURSOR_PATH "$SNAP/data-dir/icons"
 
-# Mesa Libs for OpenGL support
-append_dir LD_LIBRARY_PATH "$SNAP_DESKTOP_RUNTIME/usr/lib/$ARCH/mesa"
-append_dir LD_LIBRARY_PATH "$SNAP_DESKTOP_RUNTIME/usr/lib/$ARCH/mesa-egl"
-
-# Tell libGL and libva where to find the drivers
-export LIBGL_DRIVERS_PATH="$SNAP_DESKTOP_RUNTIME/usr/lib/$ARCH/dri"
-append_dir LD_LIBRARY_PATH "$LIBGL_DRIVERS_PATH"
-append_dir LIBVA_DRIVERS_PATH "$SNAP_DESKTOP_RUNTIME/usr/lib/$ARCH/dri"
-
-# Set where the VDPAU drivers are located
-export VDPAU_DRIVER_PATH="/usr/lib/$ARCH/vdpau/"
-if [ -e "/var/lib/snapd/lib/gl/vdpau/libvdpau_nvidia.so" ]; then
-  export VDPAU_DRIVER_PATH="/var/lib/snapd/lib/gl/vdpau"
-  if [ "$__NV_PRIME_RENDER_OFFLOAD" = 1 ]; then
-    # Prevent picking VA-API (Intel/AMD) over NVIDIA VDPAU
-    # https://download.nvidia.com/XFree86/Linux-x86_64/510.54/README/primerenderoffload.html#configureapplications
-    unset LIBVA_DRIVERS_PATH
+# Unless told to defer to the gpu interface
+# https://mir-server.io/docs/the-gpu-2404-snap-interface
+if [ "%WITH_GRAPHICS%" != "false" ]; then
+  # Give XOpenIM a chance to locate locale data.
+  # This is required for text input to work in SDL2 games.
+  export XLOCALEDIR="$SNAP_DESKTOP_RUNTIME/usr/share/X11/locale"
+
+  # Mesa Libs for OpenGL support
+  append_dir LD_LIBRARY_PATH "$SNAP_DESKTOP_RUNTIME/usr/lib/$ARCH/mesa"
```

Comment:
I'm not sure about this, probably I'm wrong, but... shouldn't this point to the GPU snap mount point, if the other patches are merged?

Reply:
No, this is the "legacy" case, I suppose we could improve what the parameter does...

Maybe `GPU_SNAP=mesa-2404` and then `if [ -z "%GPU_SNAP%" ]` so it's clearer which case we're handling?

It would probably be best to simply drop that section when using the GPU SNAP, but I didn't want to interfere too much with existing things.

Reply:
And let me clarify - in the default, "legacy" case, or when `WITH_GRAPHICS` is `true`, that means we **want** the desktop launcher to handle graphics - but  not, when `WITH_GRAPHICS=false`.